flink-staging/flink-table/src/main/scala/org/apache/flink/api/table/codegen/ExpressionCodeGenerator.scala

```diff
@@ -202,6 +210,23 @@ abstract class ExpressionCodeGenerator[R](
           """.stripMargin
         }
 
+      case expressions.Literal(dateValue: Date, DATE_TYPE_INFO) =>
+        val dateName = s"""date_${dateValue.getTime}"""
+        val dateStmt = s"""static java.util.Date $dateName
+             |= new java.util.Date(${dateValue.getTime});""".stripMargin
+        reusableStatements.add(dateStmt)
+
+        if (nullCheck) {
+          s"""
+            |val $nullTerm = false
```

Comment:
These are not tested anywhere. We should maybe add tests that also verify the Code Generator when `nullCheck` is `true`.